docs/machine-learning/tutorials/phone-calls-anomaly-detection.md

```diff
@@ -0,0 +1,261 @@
+
+    [!code-csharp[LoadData](./snippets/phone-calls-anomaly-detection/csharp/Program.cs#LoadData "loading dataset")]
+
+    The [LoadFromTextFile()](xref:Microsoft.ML.TextLoaderSaverCatalog.LoadFromTextFile%60%601%28Microsoft.ML.DataOperationsCatalog,System.String,System.Char,System.Boolean,System.Boolean,System.Boolean,System.Boolean%29) defines the data schema and reads in the file. It takes in the data path variables and returns an `IDataView`.
+
+## Time series anomaly detection
+
+Time series anomaly detection is the process of detecting time-series data outliers; points on a given input time-series where the behavior isn't what was expected, or "weird". These anomalies can be typical indicative of some events of interest in the problem domain: a cyber-attack on user accounts, power outage, bursting RPS on a server, memory leak, etc.
+
+To find anomaly on time series, one should first determine the period of the series. Then, the time series can be decomposed into several components as `Y = T + S + R`, where `Y` is the original series, `T` is the trend component, `S` is the seasonal componnent and `R` is the residual component of the series. This step is called [decomposition](https://en.wikipedia.org/wiki/Decomposition_of_time_series). Finally, detection is performed on the residual component to find the anomalies. In ML.NET, The SR-CNN algorithm is an advanced and novel algorithm that is based on Spectral Residual (SR) and Convolutional Neural Network(CNN) to detect anomaly on time-series( Refer to [this](https://arxiv.org/pdf/1906.03821.pdf) paper for more details on this algorithm).
+
+In this tutorial, you will see that these procedures can be completed in ML.net with simply two function invokes.
+
+## Detect Period
+
+In the first step, we invoke the `DetectSeasonality` function to determine the period of the series.
+
+### Create the DetectPeriod() method
+
+The `DetectPeriod()` method:
+
+1. Create the `DetectPeriod()` method, just after the `Main()` method, using the following code:
+
+    ```csharp
+    static void DetectPeriod(MLContext mlContext, IDataView phoneCalls)
+    {
+
+    }
+    ```
+
+2. Use the [DetectSeasonality](xref:Microsoft.ML.TimeSeriesCatalog.DetectSeasonality) function to detect period. Add it to the `DetectPeriod()` method with the following code:
+
+    [!code-csharp[DetectSeasonality](./snippets/phone-calls-anomaly-detection/csharp/Program.cs#DetectSeasonality)]
+
+3. Display the period value by adding the following as the next line of code in the `DetectPeriod()` method:
+
+    [!code-csharp[DisplayPeriod](./snippets/phone-calls-anomaly-detection/csharp/Program.cs#DisplayPeriod)]
+
+4. Add the following call to the `DetectPeriod()`method in the `Main()` method:
+
+    [!code-csharp[CallDetectPeriod](./snippets/phone-calls-anomaly-detection/csharp/Program.cs#CallDetectPeriod)]
+
+
+### Period detection results
+
+Your results should be similar to the following.
```

Comment:
Is the user supposed to run the application at this point? If so, I would specify it here.

```suggestion
Run the application. Your results should be similar to the following.
```